torch/csrc/jit/passes/constant_propagation.cpp

```diff
@@ -36,6 +37,9 @@ std::vector<IValue> runNode(Node* n) {
     if (v.isTensor()) {
       auto t = std::move(v).toTensor();
       if (t.defined()) {
+        if (t.requires_grad()) {
+          throw c10::Error("Can't insert requires grad as constant", "");
```

Comment:
Throwing here has no meaning -- how is a user supposed to fix that? It should probably be an assertion.

Reply:
This isn't user-visible, it will be caught by constant propagation